### PR TITLE
Dashboard redesign

### PR DESCRIPTION
**Pull request by hbagarich**
Added grouping options to main Dashboard. Servers can now be shown grouped by: Smart (the logical group they belong), Status (by status), Type and None (the servers are all shown without grouping). 
This option is propagated along all dashboards (Compact, Tile, Grid).
Note this is a breaking change, the Props Dashboard accepts were changed. Additionally the Horizontal Compact dashboard has been removed.

### CODE REVIEW

File: src/components/CompactServer/CompactServer.tsx [hunk 1]
```diff
@@ -22,6 +27,7 @@ export class CompactServer extends React.PureComponent<ICompactServerProps, any>
                 onMouseEnter={this.props.onMouseEnter}
                 onMouseLeave={this.props.onMouseLeave}
                 onClick={this.onclick}
+                ref={(element) => this.ContainerElement = element}
```
Comment (MatijaA) [inline src/components/CompactServer/CompactServer.tsx]: Arrow function will force re-render when nothing has changed

File: src/components/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -80,39 +229,41 @@ export class Dashboard extends React.PureComponent<IDashboardProps, any> {
                         filter={filter}
```
Comment (MatijaA) [inline src/components/Dashboard/Dashboard.tsx]: Shouldn't ActiveDashboard.CompactHorizontal be removed now that the Horizontal Compact dashboard has been removed?

Comment (hbagarich) [inline src/components/Dashboard/Dashboard.tsx]: yup, removed compact vertical completely from library.

File: src/components/CompactServer/CompactServer.tsx (hunk 1 above)
Comment (hbagarich) [inline src/components/CompactServer/CompactServer.tsx]: Changed it to a binded callback

**Pull request merged**